model/src/main/java/uk/gov/service/payments/commons/model/CardExpiryDate.java

```diff
@@ -25,7 +25,13 @@ private CardExpiryDate(String expiryDate) {
         this.month2Digits = matcher.group(1);
         this.year2Digits = matcher.group(2);
     }
-
+    
+    private CardExpiryDate(YearMonth expiryDate) {
+        int month = expiryDate.getMonthValue();
+        this.month2Digits = (month < 10) ? ("0" + month) : String.valueOf(month);
+        this.year2Digits = String.valueOf(expiryDate.getYear()).substring(2,4);
+    }
```

Comment:
This will mean that something like this is possible…

```java
var yearMonth = YearMonth.of(1990, 1); // 2124-01
var cardExpiryDate = CardExpiryDate.valueOf(yearMonth);
var fourDigitYear = cardExpiryDate.getFourDigitYear(); // returns 2024
var roundTrippedYearMonth = cardExpiryDate.toYearMonth(); // returns 2024-01
```
… which is a little odd.

I can think of 3 ways to handle this:

1. Leave the code as it is. It’s unlikely to actually be an issue for more than 70 years.
2. Add some validation to only allow years that are between 2000 and 2099 (inclusive).
3. Change `CardExpiryDate` so internally it stores the year as four digits (converting two-digit years to four digits if necessary) and make the `getFourDigitYear()` return years outside of the 2000–2099 range when appropriate.

While 3 appeals to me, it’s probably more work to implement than we’d want right now, so I’d be happy with 2. You are welcome to persuade me that 1 is fine, however.

Reply:
Thanks - I guess (2) is the best option. I had assumed that in the past had gone with option (1) as the get4DigitYear method will always add 20 to the two-digit date, but I suppose the Pay Team circa 2100 won't thank us for that.

Reply:
Yes, we’d kinda gone with 1 but because you could only construct a `CardExpiryDate` from a two-digit year (which was assumed to begin with 20), there wasn’t the round-tripping weirdness we have if we allow construction from a `YearMonth`.